homeassistant/components/zimi/light.py

```diff
@@ -0,0 +1,138 @@
+"""Platform for light integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from zcc import ControlPoint
+from zcc.device import ControlPointDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+
+# Import the device class from the component that you want to support
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ZimiConfigEntry
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ZimiConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Zimi Light platform."""
+
+    api: ControlPoint = config_entry.runtime_data
+
+    async_add_entities(
+        ZimiLight(device, api)
+        for device in filter(lambda light: light.type != "dimmer", api.lights)
+    )
+    async_add_entities(
+        ZimiDimmer(device, api)
+        for device in filter(lambda light: light.type == "dimmer", api.lights)
+    )
+
+
+class ZimiLight(LightEntity):
+    """Representation of a Zimi Light."""
+
+    def __init__(self, light: ControlPointDevice, api: ControlPoint) -> None:
+        """Initialize a ZimiLight."""
+
+        self._attr_unique_id = light.identifier
+        self._attr_should_poll = False
+        self._attr_has_entity_name = True
+        self._attr_name = None
```

Comment:
move the static things out of the constructor

Reply:
Done https://github.com/home-assistant/core/pull/129876/commits/dd540ab1e4f551074a3b9ad03bf7d8316d661b91